### DISCUSSION TITLE
actuall speed of classification and restart of services

### DISCUSSION THREAD

**Discussion by spicemint** [nextcloud/recognize#129] [Q&A] [answered]
hi :) 
my server has 2 dedicated Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz and 64 gbit of ram. I have about 800.000 pics at this moment, which would mean with 100 pics in 10 mins it will take like 2 months non-stop recognizing, is there any way to determine the actual speed or time to completion? and what happens when I need to restart my machine, will it re-start the recognize process from start or continue where it stopped?

**Comment by bugsyb** [comment]
Wondering if you have any stats if the process has been completed?
For very same issue of large collection, I've looked into using GPU and multiple nodes.
Running on Intel(R) Xeon(R) CPU E3-1505M v6 @ 3.00GHz wasn't satisfactory and thanks to @marcelklehr , GPU has been now added. For simplicity of deployment, there's also Docker container which can be built https://github.com/bugsyb/recognize_docker

What I don't know yet is if adding more processing nodes, speeds up the process or these would end up processing same files.

**Reply by marcelklehr** [reply, marked as answer]
> adding more processing nodes, speeds up the process or these would end up processing same files

Adding more processing nodes can speed up the process only if the files are distributed over multiple storages as recognize currently scales per storage

**Comment by marcelklehr** [comment]
> what happens when I need to restart my machine, will it re-start the recognize process from start or continue where it stopped?

Recognize knows which files have been completed and will not need to start from the beginning
